inc/mailcollector.class.php

```diff
@@ -2019,11 +2019,20 @@ public function getDecodedContent(\Laminas\Mail\Storage\Part $part) {
             break;
       }
 
-      $contentType = $part->getHeader('contentType');
-      if ($contentType instanceof \Laminas\Mail\Header\ContentType
-          && preg_match('/^text\//', $contentType->getType())
-          && mb_detect_encoding($contents) != 'UTF-8') {
-         $contents = Toolbox::encodeInUtf8($contents, $contentType->getEncoding());
+      try {
+         $contentTypePart = $part->getHeader('contentType');
+         $contentType = $contentTypePart->getType();
+      } catch (\Laminas\Mail\Storage\Exception\InvalidArgumentException $e) {
+         //no ContentType header, switch to acceptable default
+         $contentType = "text/plain";
+      } finally {
+         $encoding = mb_detect_encoding($contents);
+         if (preg_match('/^text\//', $contentType) && $encoding != 'UTF-8') {
```

Comment:
I do not know how `mb_detect_encoding` would act with images or other binary docs, maybe doing this would prevent performances/memory issues with huge attached files.
```suggestion
         if (preg_match('/^text\//', $contentType) && ($encoding = mb_detect_encoding($contents)) != 'UTF-8') {
```